File: 6. Stacks/copyStack.java
package stacks;
import java.util.Stack;
import java.util.Scanner;

public class copyStack {

	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
		Stack<Integer> st = new Stack<>();
		System.out.println("Enter the number of elements : ");
		int size = s.nextInt();
		for(int i = 0; i<size;i++)
		{
			System.out.println("Enter the element for index "+i+" : ");
			int data = s.nextInt();
			st.push(data);
		}
		System.out.println("Stack 1 : "+st);
		
		//reverse-stack
		Stack<Integer> st2 = new Stack<>();
		//Approach-1
//		while(st.size()>0)
//		{
//			int elem = st.peek();
//			st2.push(elem);
//			st.pop();
//		}
		while(st.size()>0)
		{
			st2.push(st.pop());
		}
		System.out.println("Reversed Stack : "+st2);
		//final-stack
		Stack<Integer> st3 = new Stack<>();
		while(st2.size()>0)
		{
			st3.push(st2.pop());
		}
		System.out.println("Final Stack : "+st3);

	}

}

File: Yash/C++/DSA/4. Stacks/copyStack.cpp
#include <iostream>
#include <stack>
using namespace std;

int main()
{
    stack <int> st1;
    int data = 0;
    cout<<"Enter data(press -1 to stop entering ) : "<<endl;
    cin>>data;
    while(data!= -1)
    {
        st1.push(data);
        cin>>data;
    }

    //copy-stack
    stack <int> st2;

    while(!(st1.size() == 0))
    {
        int element = st1.top();
        st2.push(element);
        st1.pop();
    }

    //printing st2;
    while(!(st2.size() == 0))
    {
        cout<<st2.top()<< " ";
        st2.pop();
    }
}